lib/srv/db/streamer.go

```diff
@@ -78,18 +76,7 @@ func (s *Server) newStreamer(ctx context.Context, sessionID string, recConfig ty
 	s.log.Debugf("Using async streamer for session %v.", sessionID)
 	uploadDir := filepath.Join(
 		s.cfg.DataDir, teleport.LogsDir, teleport.ComponentUpload,
-		libevents.StreamingLogsDir, apidefaults.Namespace)
-	// Make sure the upload dir exists, otherwise file streamer will fail.
-	_, err := utils.StatDir(uploadDir)
-	if err != nil && !trace.IsNotFound(err) {
-		return nil, trace.Wrap(err)
-	}
-	if trace.IsNotFound(err) {
-		s.log.Debugf("Creating upload dir %v.", uploadDir)
-		if err := os.MkdirAll(uploadDir, 0755); err != nil {
```

Comment:
Are these directories created elsewhere now, or was this always redundant check?

Reply:
The creation was moved to the component that actually depends on the directory being there.

https://github.com/gravitational/teleport/pull/19011/files#diff-19d75ef44d31857c535a22a5eefc8423ad461f311575142e0777c876c27cb254R95-R100